### PR TITLE
DSD-944: Fixing Card layout when isAlignedRightActions is true

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-944](https://jira.nypl.org/browse/DSD-944)

## This PR does the following:

- Updates CSS for the "body" and "actions" section elements in the `Card` component when `isAlignedRightActions` is set to `true`.

## How has this been tested?

Storybook. I tested this with different elements and the different responsive views. Let me know if the changes you suggested are enough -- I only added them to the mobile view since adding them all the time did not render the desktop layout correctly.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr980-l0z23aejgrtbuwl8dxrjv9o3zit4mjr9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626adb7aef17c6f569235c70

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr980-l0z23aejgrtbuwl8dxrjv9o3zit4mjr9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626adb7aef17c6f569235c70

**Comment by bigfishdesign13** [comment]
@EdwinGuzman Which configurations were showing problems when `flexBasis: “100%”` was applied all the time?  Without that attribute, there are sizing problems when the text content is short.  This is easiest to see when the CardActions are set to the right.  Pull up that story in Storybook and use the browser dev tools to set the content to only a few words.  When you do that, the card body region will contract to the size of the text and the card actions region will slide to the left based on the size of the card body region.

**Comment by EdwinGuzman** [comment]
I see the problem. It helps when the actions section has a max-width rather than just width of 180px. Check the update.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr980-l0z23aejgrtbuwl8dxrjv9o3zit4mjr9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626adb7aef17c6f569235c70

**Comment by bigfishdesign13** [comment]
@EdwinGuzman Yes, this now works well for the `actions on right` variant, but there are also sizing issues on for the standard row layout.  To see what I mean, do the same thing with the text for the initial `Card` story.  The actions region will shrink to be the same size as the text.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr980-l0z23aejgrtbuwl8dxrjv9o3zit4mjr9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626adb7aef17c6f569235c70

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr980-l0z23aejgrtbuwl8dxrjv9o3zit4mjr9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626adb7aef17c6f569235c70

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr980-l0z23aejgrtbuwl8dxrjv9o3zit4mjr9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626adb7aef17c6f569235c70

File: src/theme/components/card.ts [hunk 1]
```diff
@@ -112,12 +114,14 @@ const Card = {
...
-        marginLeft: ["0", "0", "m"],
-        marginTop: ["xs", "xs", "0"],
+        flexShrink: { base: isAlignedRightActions ? "0" : null, md: "0" },
+        marginLeft: { base: "0", md: "m" },
+        marginTop: { base: "xs", md: "0" },
+        maxWidth: { base: "100%", md: "180px" },
       },
       body: {
         display: { md: "block" },
+        flexBasis: { sm: isAlignedRightActions ? "100%" : null },
```
Comment (bigfishdesign13) [inline src/theme/components/card.ts]: Changing `isAlignedRightActions` to `isRow` seems to fix the width problems for all `row` configurations.

File: src/theme/components/card.ts [hunk 2]
```diff
@@ -112,12 +114,14 @@ const Card = {
       flexFlow: "column wrap",
       textAlign: isCentered ? "center" : null,
       actions: {
-        width: ["100%", "100%", "180px"],
-        marginLeft: ["0", "0", "m"],
-        marginTop: ["xs", "xs", "0"],
+        flexShrink: { base: isAlignedRightActions ? "0" : null, md: "0" },
+        marginLeft: { base: "0", md: "m" },
+        marginTop: { base: "xs", md: "0" },
+        maxWidth: { base: "100%", md: "180px" },
```
Comment (bigfishdesign13) [inline src/theme/components/card.ts:120]: Adding `width: "100%”` to this block makes the actions element go to full width in a responsive view.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Some minor changes to what you have will fix the width problems in this and other configurations.

**Comment by EdwinGuzman** [comment]
Updated.

File: src/theme/components/card.ts [hunk 3]
```diff
@@ -112,12 +114,15 @@ const Card = {
...
+        width: "100%",
       },
       body: {
```
Comment (bigfishdesign13) [inline src/theme/components/card.ts:123]: Ugh.  I found one more configuration that still needs to be cleaned up.  Please add the following to the styles for `body`:
```
width: { base: "100%", md: "auto" },
```

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Sorry, one more update.  So many configs for this component.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.  I think that is it!

**Pull request merged**